gpiozero/rotary_encoder.py

```diff
@@ -0,0 +1,253 @@
+    and the middle pin to a ground pin. Alternatively, connect the middle pin to the
+    3V3 pin, then set *pull_up* to ``False`` in the :class:`RotaryEncoder` constructor.
+
+    The following example will print a Rotary Encoder change direction::
+
+        from gpiozero import RotaryEncoder
+
+        def change(value):
+            if value > 0:
+                print("clockwise")
+
+            else:
+                print("counterclockwise")
+
+        rotary = RotaryEncoder(13, 19)
+        rotary.when_rotated = change
+
+    Based in Pigpio implementation `Pigpio implementation <http://abyz.co.uk/rpi/pigpio/examples.html#Python_rotary_encoder_py>`_
+    and `Paul Stoffregen implementation <https://github.com/PaulStoffregen/Encoder>`_
+
+    :param int pin_a:
+        An extreme GPIO pin which the RotaryEncoder is attached to. See :ref:`pin_numbering`
+        for valid pin numbers.
+
+    :param int pin_b:
+        The another extreme GPIO pin which the RotaryEncoder is attached to. See :ref:`pin_numbering`
+        for valid pin numbers.
+
+    :param bool pull_up:
+        If ``True`` (the default), the GPIO pins will be pulled high by default.
+        In this case, connect the middle GPIO pin to ground. If ``False``, 
+        the GPIO pins will be pulled low by default. In this case,
+        connect the middle pin of the RotaryEncoder to 3V3.
+    """
+
+    when_rotated = lambda *args: None
+
+    def __init__(self, pin_a, pin_b, pull_up=True):
+        self.pin_a = DigitalInputDevice(pin=pin_a, pull_up=pull_up)
+        self.pin_b = DigitalInputDevice(pin=pin_b, pull_up=pull_up)
+
+        self.pin_a.when_activated = self._pulse
+        self.pin_a.when_deactivated = self._pulse
+
+        self.pin_b.when_activated = self._pulse
+        self.pin_b.when_deactivated = self._pulse
+
+        self._old_a_value = self.pin_a.is_active
+        self._old_b_value = self.pin_b.is_active
+
+    def _pulse(self):
+        """
+        Calls when_rotated callback if detected changes
+        """
+        new_b_value = self.pin_b.is_active
+        new_a_value = self.pin_a.is_active
+
+        value = TableValues.value(new_b_value, new_a_value, self._old_b_value, self._old_a_value)
+
+        self._old_b_value = new_b_value
+        self._old_a_value = new_a_value
+
+        if value != 0:
+            self.when_rotated(value)
+
+    def close(self):
+        self.pin_a.close()
+        self.pin_b.close()
+
+    @property
+    def closed(self):
+        return self.pin_a.closed and self.pin_b.closed
+
+    @property
+    def is_active(self):
+        return not self.closed
+
+    @property
+    def value(self):
+        return None
+
+    def __repr__(self):
+        return "<gpiozero.%s object on pin_a %r, pin_b %r, pull_up=%s, is_active=%s>" % (
+                self.__class__.__name__, self.pin_a.pin, self.pin_b.pin, self.pin_a.pull_up, self.is_active)
+
+
+class TableValues:
```

Comment:
I did not understand. Python 2.7 works https://travis-ci.org/RPi-Distro/python-gpiozero/jobs/165127085